Translate  react-dom/render section (#561)
reactjs/es.reactjs.org: beta/src/content/apis/react-dom/render.md

<Pitfall>

En React 18, `render` fue reemplazado por [`createRoot`.](/apis/react-dom/client/createRoot) Utilizando `render` en React 18 se te advertirá que tu aplicación se comportará como si estuviera ejecutándose en React 17. Aprende mas [aquí.](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis)

</Pitfall>

<Intro>

> `render` renderiza una pieza de [JSX](/learn/writing-markup-with-jsx) ("React node") en un nodo del DOM del navegador.

Reviewer: ```suggestion
`render` renderiza una pieza de [JSX](/learn/writing-markup-with-jsx) ("nodo de React") en un nodo del DOM del navegador.
```
Está bien traducir términos técnicos siempre que haya traducciones «naturales» en español.

render(<App />, document.getElementById('root'));
````

> ### Renderiza el componente raíz "Root component" {/*rendering-the-root-component*/}

Author: quería dejar la traducción original por si alguien busca cómo se llama en INgles.. pero así está bien tambien jeje.

</Sandpack>

Puedes destruir los árboles renderizados con [`unmountComponentAtNode()`.](/apis/react-dom/unmountComponentAtNode)

---

> ### Actualizando el árbol renderizado {/*updating-the-rendered-tree*/}

Author: Updating no conjuga la palabra ¿? por eso pensaba que quedaba mejor Actualizando y no el verbo en infinitivo.

export default function App({counter}) {
  return (
    <>
      <h1>Hola, mundo! {counter}</h1>
      <input placeholder="Escribe algo aquí" />
    </>
  );
}
```

</Sandpack>

> No es muy común llamar a `render` varias veces. Por lo general [actualizarás el estado](/apis/react/useState) dentro de uno de los componentes.

Author: en ese caso te propongo mejor:  "Por lo general lo que debes hacer es" [ actualizar....

¿Qué opinas?.

> No es muy común llamar a `render` varias veces. Por lo general [actualizarás el estado](/apis/react/useState) dentro de uno de los componentes.

Reviewer: De acuerdo.

---

## Referencia {/*reference*/}

### `render(reactNode, domNode, callback?)` {/*render*/}

Utiliza `render` para mostrar un componente de React dentro del navegador o elemento del DOM.

```js
const domNode = document.getElementById('root');
render(<App />, domNode);
```

React mostrará `<App />`  en el `domNode`, y se encargará de gestionar el DOM dentro de él.

> Una aplicación totalmente construida con React usualmente tendrá sólo un llamado a `render` con su componente raíz "root". Una página que utiliza "sprinkles" de React para partes de la página puede tener tantas llamadas `render` como sean necesarias.

Reviewer: ```suggestion
Una aplicación totalmente construida con React tendrá usualmente una sola llamada a `render` con su componente raíz. Una página que utiliza React para partes de la página puede tener tantas llamadas a `render` como sean necesarias.
```

> [Vea los ejemplos anteriores.](#usage)

Reviewer: ```suggestion
[Mira los ejemplos anteriores.](#usage)
```

https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#usted-t%C3%BA-y-vos

#### Parámetros {/*parameters*/}

> * `reactNode`: Un *React node* que tu quieras mostrar. Por lo general se trata de un fragmento o pieza de JSX `<App />`, pero también puedes pasar un elemento de React "React element" construido con [`createElement()`](/apis/react/createElement), una cadena de text "string", un número "number", `null`, or `undefined`.

Reviewer: ```suggestion
* `reactNode`: Un *nodo de React* que quieras mostrar. Por lo general se trata de una pieza de JSX como `<App />`, pero también puedes pasar un elemento de React construido con [`createElement()`](/apis/react/createElement), un _string_, un número, `null`, o `undefined`.
```

> * `domNode`: Un [elemento del DOM.](https://developer.mozilla.org/en-US/docs/Web/API/Element) React mostrará el `reactNode` que tú pases dentro de este elemento del DOM. Desde este momento, React administrará el DOM dentro de `domNode` y lo actualizará cuando tu árbol de React cambie.

Reviewer: ```suggestion
* `domNode`: Un [elemento del DOM.](https://developer.mozilla.org/en-US/docs/Web/API/Element) React mostrará el `reactNode` que pases dentro de este elemento del DOM. Desde este momento, React administrará el DOM dentro de `domNode` y lo actualizará cuando tu árbol de React cambie.
```

> * **Opcional** `callback`: Una función. Si se pasa, React la llamará luego de que tu componente sea colocado dentro del DOM.

Reviewer: ```suggestion
* `callback` **opcional**: Una función. Si se pasa, React la llamará luego de que tu componente sea colocado dentro del DOM.
```

> #### Retorno - Returns {/*returns*/}

Reviewer: ```suggestion
#### Retorno {/*returns*/}
```

> `render` Por lo general retorna `null`. Sin embargo, si el `reactNode` que pasas es de tipo *class component*, entonces este retornará una instancia de ese componente.

Reviewer: ```suggestion
`render` Por lo general retorna `null`. Sin embargo, si el `reactNode` que pasas es un *component de clase*, entonces retornará una instancia de ese componente.
```

https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#traducciones-comunes

#### Advertencias {/*caveats*/}

* En React 18, `render` fue reemplazado por [`createRoot`.](/apis/react-dom/client/createRoot) Por favor usa `createRoot` para React 18 y versiones posteriores.

> * La primer ver que llamas a `render`, React limpiará todo el HTML contenido existente dentro del `domNode` antes de renderizar el componente de React dentro de este. Si tu `domNode` contiene HTML generado por React en el servidor o durante la construcción, usa [`hydrate()`](/apis/react-dom/hydrate) en lugar de, ya que este adjunta los manejadores de evento al HTML existente.

Reviewer: ```suggestion
* La primera vez que llamas a `render`, React limpiará todo el contenido HTML existente dentro del `domNode` antes de renderizar el componente de React dentro de este. Si tu `domNode` contiene HTML generado por React en el servidor o durante la compilación, usa en su lugar [`hydrate()`](/apis/react-dom/hydrate), ya que este adjunta los manejadores de eventos al HTML existente.
```

> * Si llamas a `render` en el mismo `domNode` mas de una vez, React actualizará el DOM según sea necesario para reflear el mas reciente JSX que hayas pasado. React decidirá que partes del DOM pueden ser reutilizadas y cuáles necesitan se recreadas ["emparejándolas"](/learn/preserving-and-resetting-state) con el árbol previamente renderizado. Llamar a `render` en el mismo `domNode` nuevamente es similar a llamar a la función [`set` ](/apis/react/useState#setstate) en el componente raíz "root": React evita actualizaciones innecesarias del DOM.

Reviewer: ```suggestion
* Si llamas a `render` en el mismo `domNode` más de una vez, React actualizará el DOM según sea necesario para reflejar el JSX más reciente que hayas pasado. React decidirá qué partes del DOM se pueden reutilizar y cuáles necesitan ser recreadas ["haciendo una comparación"](/learn/preserving-and-resetting-state) con el árbol previamente renderizado. Llamar de nuevo a `render` en el mismo `domNode` es similar a llamar a la función [`set` ](/apis/react/useState#setstate) en el componente raíz: React evita actualizaciones innecesarias del DOM.
```

> * Si tu aplicación está totalmente construida con React, es probable que sólo tengas un llamado a `render` en tu aplicación. (Si usas un framework, éste puede hacer el llamado por ti.) Cuando quieras renderizar un fragmento de JSX en un lugar diferente del árbol DOM que no sea hijo de tu componente (por ejemplo, una modal o un tooltip), usa [`createPortal`](/apis/react-dom/createPortal) en lugar de `render`.

Reviewer: ```suggestion
* Si tu aplicación está totalmente construida con React, es probable que tengas una sola llamada a `render` en tu aplicación. (Si usas un framework, puede que haga esta llamada por ti.) Cuando quieras renderizar un fragmento de JSX en un lugar diferente del árbol del DOM que no sea hijo de tu componente (por ejemplo, un modal o un _tooltip_), usa [`createPortal`](/apis/react-dom/createPortal) en lugar de `render`.
```

Para mantener la consistencia con la traducción de _«call»_ en este tipo de contextos en las demás páginas de la documentación deberíamos usar «llamada».
